### DISCUSSION TITLE
What can I do if Nextcloud shows `Update needed`?

### DISCUSSION THREAD

**Discussion by szaimen** [nextcloud/all-in-one#1787] [Wiki]
See https://github.com/nextcloud/all-in-one/discussions/2726

**Comment by TheNomad11** [comment]
This happened to me just an hour ago, but this command did not work for me. I got this famous message after running the command:

>Nextcloud or one of the apps require upgrade - only a limited number of commands are available You may use your browser or the occ upgrade command to do the upgrade

As you wrote - it was an app upgrade that (again) broke Nextcloud, this time it was Related Resouces, so I disabled it by running

`sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ app:disable related_resources`

and Nextcloud was back, alive and kicking!

**Reply by szaimen** [reply]
cc @foxhoundv

**Reply by foxhoundv** [reply]
This was probably the issue. I did run an update for this in the morning, but things were okay for a while. I was off nextcloud for about an hour and when I came back I had the issue. LOL, @szaimen I feel a little bummed that this stuff isn't affecting you too. Isn't related resources installed normally with Nextcloud AIO?

**Reply by szaimen** [reply]
Yes it is but It seems like I have not installed the latest related ressources update 1.0.4 yet on my test instance. Only 1.0.3 currently.

So I will probably run into the issue tomorrow during the auto-update. 

But I fear there is nothing I can do to fix this. At least I reported this now so that the maintainer can have a look.

**Reply by foxhoundv** [reply]
Well after running the update command, it shows my "related resource" app as v1.0.4. So i don't know if the update command made it work somehow??

**Reply by szaimen** [reply]
yeah, so probably 1.0.4 seems to be the broken release. I suppose I will get the same problem tomorrow.

**Reply by foxhoundv** [reply]
Just for more information on this, here is the CLI output when the update command was invoked...
![image](https://user-images.githubusercontent.com/16512250/213007520-d39f63eb-4ed8-4a6e-b4eb-661408ea65cd.png)

**Reply by szaimen** [reply]
Yes, thanks

**Reply by szaimen** [reply]
I'll try to reproduce the issue tomorrow.

**Reply by szaimen** [reply]
Fix is in https://github.com/nextcloud/server/pull/36257

**Comment by pktiuk** [comment]
> However the only real solution is installing less apps that potentially could trigger this.

Is here any other way of preventing these triggers? (apart from installing less apps)    
Can I disable automatic updates of apps?

**Reply by szaimen** [reply]
> Is here any other way of preventing these triggers? (apart from installing less apps)

No

**Comment by iGadget** [comment]
I just got this when trying to first update the Calendar app (which resulted in an error), then trying to re-enable SnappyMail (which was for some reason disabled after a Nextcloud server update), which resulted in another error, and finally trying to remove SnappyMail (since I wasn't using it anyway and figured this would perhaps 'fix' the issue faster than trying to debug it). After that, I got the "Update needed" page.
Running the docker command as provided resulted in the Calendar app being updated and my server resuming normal operation. SnappyMail was still present (and disabled), but I was then able to remove it without issues.

**Comment by WDMdanila** [comment]
Same for me, update of Calendar resulted in `Update Needed` screen. Logs from update command:

```
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Setting log level to debug
Turned on maintenance mode
Updating database schema
Updated database
Disabled incompatible app: nextcloud-aio
Updating <calendar> ...
Updated <calendar> to 4.4.0
Starting code integrity check...
Finished code integrity check
Update successful
Turned off maintenance mode
Resetting log level
```

Maybe problem is related to that message: `Disabled incompatible app: nextcloud-aio`?

Running `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ upgrade` fixed everything

**Reply by szaimen** [reply]
> `Disabled incompatible app: nextcloud-aio`?

No, the problem is:

> The bug is reported to https://github.com/nextcloud/calendar/issues/5191  and will be fixed with https://github.com/nextcloud/calendar/pull/5295

From https://github.com/nextcloud/all-in-one/discussions/2716#discussioncomment-6118613

**Reply by szaimen** [reply]
btw, @WDMdanila in your case you need to re-enable the nextcloud-aio app: `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ app:enable nextcloud-aio --force`

**Reply by WDMdanila** [reply]
Thank you!

Funny enough but it looks like `Nextcloud All-in-One` app is not disabled - I can see it on a list of active apps, as well as have buttons which are created by app (at least I assume so)
![image](https://github.com/nextcloud/all-in-one/assets/24762845/3c00d9a8-900b-4912-b1ee-8e01fd811b6a)

**Reply by szaimen** [reply]
> Funny enough but it looks like `Nextcloud All-in-One` app is not disabled - I can see it on a list of active apps, as well as have buttons which are created by app (at least I assume so)

Interesting. Then it is probably fine.

**Comment by Moonbase59** [comment]
Thanks @szaimen for pointing me here. Nextcloud up and running again, but the Admin Settings check now reports:

```
Security & setup warnings

It's important for the security and performance of your instance that everything is configured correctly. To help you with that we are doing some automatic checks. Please see the linked documentation for more information.
There are some errors regarding your setup.

    Some files have not passed the integrity check. Further information on how to resolve this issue can be found in the [documentation ↗](https://docs.nextcloud.com/server/26/go.php?to=admin-code-integrity). (List of invalid files… / Rescan…)

Please double check the [installation guides ↗](https://docs.nextcloud.com/server/26/go.php?to=admin-install), and check for any errors or warnings in the log.

Check the security of your Nextcloud over [our security scan ↗](https://scan.nextcloud.com/).
```

Clicking on "List of invalid files" shows:

```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => 4b0a99fbab7bc4824176101150bd0ab2b553e128a390e0f872580db4541462b31aad13099fd41533b6695d1b185dabca26a45e9a78227cfb9479eafa439c7bff
                            [current] => a8e818876a7eff2a2137601eafc6c7d96b99cbf27d94ff9bb59eb237bdab92076ecc2828242bb37551c9c8b6772c0718887bd49cee5b49b3be48d3b9ecaa558f
                        )

                )

        )

)
```

Running AIO 6.1.0 beta.

**Reply by szaimen** [reply]
Hi, this is nothing to worry about: you can just ignore it.

It usually should get fixed automatically with the next Nextcloud update which is planned for beginning of July.

**Reply by Moonbase59** [reply]
Okay, thanks a bundle!

**Comment by Chrisrummy** [comment]
I edited the config.php file to look like this.

 'upgrade.disable-web' => false,
  'maintenance' => false,

Then refreshed the web page and was presented with an update prompt.  Now my app is back up.
